### PR TITLE
Feature/input manager

### PR DESCRIPTION
**Pull request by Korotsu**
+Add input Manager & input Component.

### CODE REVIEW

**Review by Renardjojo — APPROVED** [review]

**Review by ThomasProg — CHANGES_REQUESTED** [review]
I want GPEngine.vcxproj, not Test.vcxproj.

File: engine/include/Engine/Core/Input/Cursor.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ *	found in the top-level directory of this distribution.
+ */
+#pragma once
+#include "GPM/Vector2.hpp"
+
+namespace GPE
+{
+class Cursor
+{
+public:
```
Comment (ThomasProg) [inline engine/include/Engine/Core/Input/Cursor.hpp]: No need to assign them to default.

File: engine/include/Engine/Core/Input/Cursor.hpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    Cursor()  = default;
+    ~Cursor() = default;
+
+public:
+    GPM::Vec2 position = {300, 450};
```
Comment (ThomasProg) [inline engine/include/Engine/Core/Input/Cursor.hpp]: Why 300 and 450 ? Where are these values coming from ?

File: engine/src/InputManagerGLFW.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            if (it2->second == EKeyMode::KEY_PRESS && it->second == true)
+            {
+                input->m_inputComponents[i]->fireAction(action);
+                input->m_stateMap[key] = false;
+            }
+
+            else if (it2->second == EKeyMode::KEY_REPEAT && it->second == true)
+            {
+                input->m_inputComponents[i]->fireAction(action);
+            }
```
Comment (ThomasProg) [inline engine/src/InputManagerGLFW.cpp]:
```suggestion
            if (it->second == true)
            {
                  switch (it2->second)
                  {
                         case  EKeyMode::KEY_PRESS:
                                          input ->m_inputComponents[i]->fireAction(action);
                                          input->m_stateMap[key] = false;
                                          break;
                         case  EKeyMode::KEY_REPEAT:
                                          input->m_inputComponents[i]->fireAction(action);
                                          break;
                  }
            }
```

**Review by ThomasProg — APPROVED** [review]

File: engine/include/Engine/Core/Input/Cursor.hpp (hunk 2 above)
Comment (Korotsu) [reply engine/include/Engine/Core/Input/Cursor.hpp]: The half-size of the window but the valor doesn't matter

**Pull request merged**